File: frontend/job-search-ui/src/app/models/job-opportunity.ts
export type RemoteType = 'Unknown' | 'Remote' | 'Hybrid' | 'Onsite';

export type ApplicationStatus =
  | 'Found'
  | 'Interested'
  | 'Applied'
  | 'Interviewing'
  | 'Rejected'
  | 'Archived';

export interface JobOpportunity {
  id: string;
  company: string;
  title: string;
  location: string | null;
  remoteType: RemoteType;
  url: string | null;
  description: string | null;
  status: ApplicationStatus;
  fitScore: number | null;
  dateFound: string;
  dateApplied: string | null;
}

export interface CreateJobOpportunityRequest {
  company: string;
  title: string;
  location: string | null;
  remoteType: RemoteType;
  url: string | null;
  description: string | null;
  fitScore: number | null;
}

export const remoteTypes: readonly RemoteType[] = ['Unknown', 'Remote', 'Hybrid', 'Onsite'];

export const applicationStatuses: readonly ApplicationStatus[] = [
  'Found',
  'Interested',
  'Applied',
  'Interviewing',
  'Rejected',
  'Archived'
];

File: backend/JobSearch.Domain/Entities/JobOpportunity.cs
using JobSearch.Domain.Enums;

namespace JobSearch.Domain.Entities;

public sealed class JobOpportunity
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public required string Company { get; set; }
    public required string Title { get; set; }
    public string? Location { get; set; }
    public RemoteType RemoteType { get; set; } = RemoteType.Unknown;
    public string? Url { get; set; }
    public string? Source { get; set; }
    public string? ExternalId { get; set; }
    public DateTimeOffset? LastSeenAt { get; set; }
    public string? Description { get; set; }
    public ApplicationStatus Status { get; set; } = ApplicationStatus.Found;
    public int? FitScore { get; set; }
    public FitScoreResult? FitScoreResult { get; set; }
    public string? GeneratedRecruiterMessage { get; set; }
    public DateTimeOffset DateFound { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? DateApplied { get; set; }
}
